code/game/objects/items/weapons/storage/RPED.dm

```diff
@@ -71,3 +71,63 @@
 		new /obj/item/weapon/stock_parts/capacitor/adv(src)
 	for(var/i in 1 to 8)
 		new /obj/item/weapon/stock_parts/console_screen(src)
+
+//Takes a tier 1 stock part path and a target rating
+//Returns a part object
+/proc/part_subtype(var/basepath, var/target)
+	for(var/path in subtypesof(basepath))
+		var/obj/item/weapon/stock_parts/SP = new path
+		if(SP.rating == target)
```

Comment:
Why can't you `initial` for that comparison operation, and only create the object if needed ?
```dm
for(var/path in subtypesof(basepath))
		var/obj/item/weapon/stock_parts/SP = path
		if(initial(SP.rating) == target)
               return new SP
```